server/scores.js

```diff
@@ -15,16 +21,54 @@ const connectionPromise = MongoClient
   .connect(mongoUrl, { promiseLibrary: Promise, useNewUrlParser: true })
   .then(client => client.db().collection('scores'))
 
+function _validateScore (score) {
+  let retError = { 'status': 'ok', 'errors': '' }
+
+  if (typeof score.teamNumber !== 'number' || score.match == null || score.score == null) {
+    retError.status = 'loud-fail'
+    if (typeof score.teamNumber !== 'number') { retError.errors += 'team number ' }
+    if (score.score == null) { retError.errors += 'score ' }
+    if (score.match == null) { retError.errors += 'match ' }
+  }
+
+  if (score.signature.isEmpty) {
+    retError.errors += 'signature '
+    retError.status = retError.status === 'loud-fail' ? 'loud-fail' : 'silent-fail'
+  }
+
+  return retError
+}
+
 const adminAction = authroizationMiddlware(['admin', 'scorekeeper', 'development'])
 
 router.post('/create', (req, res) => {
-  connectionPromise
-    .then(scoringCollection => scoringCollection.save(req.body))
-    .then(() => res.status(201).send())
-    .catch(err => {
-      req.logger.error(err.message)
-      res.status(500).send('A problem occoured while trying to save score.')
-    })
+  const scoreValidation = _validateScore(req.body)
+
+  if (scoreValidation.status !== 'ok') {
+    req.logger.error('Invalid score, missing ' + scoreValidation.errors + '. ' + scoreValidation.status + '.')
+  }
+
+  if (scoreValidation.status !== 'loud-fail') {
+    connectionPromise
+      .then(scoringCollection => {
+        scoringCollection.save(req.body)
+      })
+      .then(() => {
+        autoPublish.then(result => {

```

Comment:
This should happen in the validation method, as described above. It is not the responsibility of the routing method to validate or enrich the score